vernac/classes.ml

```diff
@@ -353,11 +348,12 @@ let declare_instance_program env sigma ~global ~poly id pri imps decl term termt
   let obls, constr, typ =
     match term with
     | Some t ->
-      let termtype = EConstr.of_constr termtype in
       let obls, _, constr, typ =
         Obligations.eterm_obligations env id sigma 0 t termtype
       in obls, Some constr, typ
-    | None -> [||], None, termtype
+    | None ->
+      let termtype = EConstr.to_constr sigma termtype in
```

Comment:
Why are we normalizing here now?

Reply:
We now pass an `EConstr.t` to the function; we are more lazy this way I think.